### PR TITLE
Add color to log output

### PR DESCRIPTION
**Pull request by sandlerben**

### CODE REVIEW

**Comment by sandlerben** [comment]
r @prashantv

**Comment by sandlerben** [comment]
I'm a bit perplexed by the build error `Package (golang.org/x/sys/unix) not found`

I think this is a problem with recent version(s) of `godep` because when I updated my local version of `godep` I got the same error.

See https://github.com/tools/godep/issues/377

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -6,9 +6,31 @@
...
+		{
+			"ImportPath": "github.com/mattn/go-colorable",
+			"Rev": "3dac7b4f76f6e17fb39b768b89e3783d16e237fe"
+		},
+		{
+			"ImportPath": "github.com/mattn/go-isatty",
+			"Rev": "56b76bdf51f7708750eac80fa38b952bb9f32639"
+		},
+		{
+			"ImportPath": "github.com/stretchr/testify/assert",
```
Comment (sandlerben) [inline Godeps/Godeps.json]: I'm not sure why this wasn't already in Godeps.json

**Comment by sandlerben** [comment]
Update: I fixed the `golang.org/x/sys/unix` issue by running `godep get golang.org/x/sys/unix`. For some reason `go get golang.org/x/sys/unix` hangs.

File: Godeps/Godeps.json (hunk 1 above)
Comment (prashantv) [inline Godeps/Godeps.json]: Created a separate PR for just the unrelated godeps,
https://github.com/uber/go-torch/pull/37

We should probably rebase once that change is merged

File: main.go [hunk 2]
```diff
@@ -25,12 +25,12 @@ package main
 import (
 	"fmt"
 	"io/ioutil"
-	"log"
 	"os"
 	"strings"
 
 	"github.com/uber/go-torch/pprof"
 	"github.com/uber/go-torch/renderer"
+	"github.com/uber/go-torch/torchLog"
```
Comment (prashantv) [inline main.go]: Package names should be lowercase,
https://blog.golang.org/package-names

File: main.go [hunk 3]
```diff
@@ -95,12 +94,12 @@ func runWithOptions(opts *options) error {
 	}
 
 	if opts.Print {
-		log.Print("Printing svg to stdout")
+		torchLog.Print("Printing svg to stdout")
 		fmt.Printf("%s", flameGraph)
```
Comment (prashantv) [inline main.go]: here and above, can you change the `Printf` to include a new line at the end of the format string

File: torchLog/torchLog.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/fatih/color"
+)
+
+var toRedString = color.New(color.FgRed).SprintFunc()
+var toBlueString = color.New(color.FgBlue).SprintFunc()
```
Comment (prashantv) [inline torchLog/torchLog.go]: nit, suggest a var group
```
var (
  toRedString = [...]
  toBlueString = [...]
)
```

File: torchLog/torchLog.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+func init() {
+	log.SetFlags(0) // disable default flags
+}
+
+// Fatalf wraps log.Fatalf and adds the current time and color.
+func Fatalf(format string, v ...interface{}) {
+	currentTime := time.Now().Format("15:04:05")
```
Comment (prashantv) [inline torchLog/torchLog.go]: can we make a helper function that will return the prefix given the color function.

```go
func getPrefix(level string, sprintf func(a ...interface{}) string) string {
  currentTime := time.Now().Format("15:04:05")
  return sprintf(fmt.Sprintf("%s[%s]", level, currentTime))
}
```

And then the other functions could just do
```go
prefix := getPrefix("FATA", toRedString)
```

**Comment by prashantv** [comment]
Thanks for the changes @sandlerben, left a few minor comments

File: Godeps/Godeps.json (hunk 1 above)
Comment (sandlerben) [inline Godeps/Godeps.json]: Sounds good, I will do that once it's merged!

File: torchLog/torchLog.go (hunk 5 above)
Comment (sandlerben) [inline torchLog/torchLog.go]: Great idea

File: main.go (hunk 3 above)
Comment (sandlerben) [inline main.go]: Let me know if the changes I made are what you meant

**Comment by sandlerben** [comment]
Thanks for the comments @prashantv, most things should be fixed now.

BTW, I realized why we use both `log.Print` and `fmt.Print`. Logging writes to stderr but the fmt `Print` functions write to stdout.

**Comment by prashantv** [comment]
Thanks for fixing @sandlerben, looks good to me.

I merged the Godeps change, so once you rebase, I can merge this in

**Comment by sandlerben** [comment]
@prashantv I rebased. Do you understand the build error? It seems like travis is trying to find the `torchlog` package but can't find it.

**Comment by prashantv** [comment]
The issue is that the `torchLog.go` is in the directory `torchLog` rather than `torchlog`. The case difference is tripping up travis.

OSX has a case insensitive filesystem, so what you'll want to do is something like:
```bash
mv torchLog torchlog_
mv torchlog_ torchlog
```

**Comment by sandlerben** [comment]
@prashantv I tried that but `git` does not detect that anything changed.

**Comment by prashantv** [comment]
Rename it, git commit it, then rename it back, then git commit it. After that, maybe just squash all the changes using `git rebase` into one.

**Comment by sandlerben** [comment]
@prashantv Ok awesome looks like it's working now!

**Comment by prashantv** [comment]
Sweet!

**Pull request merged**